src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1447,14 +1447,11 @@ public function isMethod($method)
      *
      * @see https://tools.ietf.org/html/rfc7231#section-4.2.1
      *
-     * @param bool $andCacheable Adds the additional condition that the method should be cacheable. True by default.
-     *
      * @return bool
      */
-    public function isMethodSafe(/* $andCacheable = true */)
+    public function isMethodSafe()
     {
-        if (!\func_num_args() || func_get_arg(0)) {
-            // setting $andCacheable to false should be deprecated in 4.1
```

Comment:
I guess this did not happen? :see_no_evil:

Reply:
Sorry, I see it's already done in 4.4: https://github.com/symfony/symfony/pull/31658

Reply:
This change seems dangerous actually. Consider this scenario:

- App was using Symfony < 3.2 (where the change was originally introduced: https://github.com/symfony/symfony/pull/20603)
- App contains `$request->isMethodSafe()` calls which expect the old (< 3.2) behaviour
- Developer upgrades app to Symfony 4.3
- The calls would return the wrong value silently after this PR

Reply:
This upgrade path is not supported - one must go through 3.4 and fix deprecations. We don't support another path.

Reply:
I'm trying to understand the rationale behind the double deprecation strategy, i.e.:

1. Deprecate not passing the `$andCacheable` argument with `false` value in `3.2`.
2. Throwing an error when not passing the `$andCacheable` argument with `false` value in `4.0`.
3. Deprecate passing the `$andCacheable` argument in ~`4.1`~`4.4`.

It seems to me like it's meant to prevent having unexpected behaviour when the deprecation is not caught during the upgrade process (and this can and does happen - it's impossible to catch all deprecations). And this PR seems to break the safety afforded by this strategy.

Reply:
The reasoning is that the deprecation is triggered in 3.4. So we *can* assume in 4.x that it's been taken into account.